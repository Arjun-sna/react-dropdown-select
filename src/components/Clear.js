import React from 'react';
import styled from '@emotion/styled';

const Clear = ({ props, state, methods }) =>
  props.clearRenderer ? (
    props.clearRenderer(props, state, methods)
  ) : (
    <ClearComponent
      className="react-dropdown-select-clear"
      tabIndex="-1"
      onClick={() => methods.clearAll()}
      onKeyPress={() => methods.clearAll()}>
      &times;
    </ClearComponent>
  );

const ClearComponent = styled.div`
  line-height: 25px;
  margin: 0 10px;
  cursor: pointer;

  :focus {
    outline: none;
  }

  :hover {
    color: tomato;
  }
`;

export default Clear;
