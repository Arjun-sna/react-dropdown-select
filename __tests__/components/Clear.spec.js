import React from 'react';
import renderer from 'react-test-renderer';

import Clear from '../../src/components/Clear';

const props = {
  parentProps: {
    clearRenderer: null
  }
};

it('<Clear/> renders correctly', () => {
  const tree = renderer.create(<Clear {...props}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
