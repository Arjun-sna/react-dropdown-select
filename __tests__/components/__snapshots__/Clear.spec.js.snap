// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clear/> component <Clear/> renders correctly 1`] = `
<div
  className="react-dropdown-select-clear css-x3l5ce-ClearComponent e11qlq5e0"
  onClick={[Function]}
  onKeyPress={[Function]}
  tabIndex="-1"
>
  Ã—
</div>
`;
